### PR TITLE
Feat/optimize image

### PR DESCRIPTION
**Pull request by kiesha-telus**
## Related issues

See related issue: https://github.com/telus/tds-community/issues/439

## Description

Component optimizes images to render by modifying Contentful asset urls. Detects browser support for webp format and converts the image to this format for faster load time. Sizes images according to screen width to further help reduce image file size at load time.

## Checklist before submitting pull request

- Commits follow our [Developer Guide](https://github.com/telus/tds-community/blob/chore/template-hints/.github/CONTRIBUTING.md)
- [x] New code is unit tested
- [ ] make sure visual and accessibility tests pass
- [x] make sure code builds

### CODE REVIEW

**Review by jraff — CHANGES_REQUESTED** [review]
1.  Please remove `shared/utils/generateId/package-lock.json`
2. Why are there updates to the Tabs and Tags snapshots?

**Comment by kiesha-telus** [comment]
the Tabs/Modal snapshots were updated because the tds-core package for core-interactive-icon was updated. let me know if i should revert this

File: packages/OptimizeImage/OptimizeImage.jsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  const [imgUrls, setImgUrls] = useState({})
+
+  useEffect(() => {
+    if (!contentfulAssetUrl.match(/.svg/g)) {
+      // Currently not all browsers support webP
+      checkWebpFeature(result => {
+        let format = ''
+        if (result) {
+          format = 'fm=webp'
+        } else if (contentfulAssetUrl.match(/.jpeg/g)) {
```
Comment (Andrew-K-Lam) [inline packages/OptimizeImage/OptimizeImage.jsx:22]: If its not a webp or a jpeg, what will it default to?  How will this handle pngs (pngs need to be supported)

**Review by Andrew-K-Lam — CHANGES_REQUESTED** [review]

File: packages/OptimizeImage/OptimizeImage.jsx (hunk 1 above)
Comment (kiesha-telus) [reply packages/OptimizeImage/OptimizeImage.jsx:22]: png's don't need further formatting and can be used as is.

**Review by Andrew-K-Lam — APPROVED** [review]

**Review by jraff — APPROVED** [review]

**Pull request merged**